cognite/processpool/processpool.py

```diff
@@ -202,14 +208,14 @@ def _job_manager_thread(self):
                 return
             self._pool[idle_procs[0]].send(job)
 
-    def submit_job(self, *args, **kwargs) -> Future:
+    def submit_job(self, *args: Any, **kwargs: Any) -> Future[Result]:
         """Submits job asynchronously, which will eventually call the `run` method in worker_class with the arguments given, all of which should be picklable."""
         if self.terminated or self.shutting_down:
             raise ProcessPoolShutDownException("Worker pool shutting down or terminated, can not submit new jobs")
-        future = Future()  # type: Future
+        future: Future[Result] = Future()
```

Comment:
Why does mypy need this explicitly? It should be able to handle inference here. Or is it for the generic?

Reply:
For the generic, and I just preferred it in front, would probably work as a comment too

Reply:
Between the name and the assignment is the way to go 👍🏻